reflex/event.py

```diff
@@ -575,12 +576,32 @@ def download(url: str | Var, filename: Optional[str | Var] = None) -> EventSpec:
         # if filename is not provided, infer it from url
         if filename is None:
             filename = url.rpartition("/")[-1]
+    
+    if url is None:
+        if filename is None:
+            filename = "download.txt"
+
+        return server_side(
+            "_download",
+            get_fn_signature(download),
+            filename=filename,
+            data=data,
+        )
+
+    if data is None:
+        return server_side(
+            "_download",
+            get_fn_signature(download),
+            url=url,
+            filename=filename,
+        )
```

Comment:
why are these multiple calls needed? if you pass `None` through the event how does it come out on the js side? if it comes as a `null`, then maybe you can change the `if` in `state.js` to check for that.

I think this function would be cleaner if there was only a single call to `server_side`.

Reply:
have changed this to use a dict that only passes the arguments through if they are not none